tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php

```diff
@@ -428,6 +450,60 @@ function ( Request $request ) {
 		$this->assertEquals( 'https://example.com/fresh.jpg', $profile['photo'] );
 	}
 
+	public function test_authorize_user__with_redirect_url_notification() {
+		$user_id = $this->factory()->user->create();
+		wp_set_current_user( $user_id );
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE, new MutableInput() );
+		$user_options = new User_Options( $context );
+		$client       = new OAuth_Client( $context, null, $user_options );
+		$this->fake_site_connection();
+
+		// Add a notification query parameter to the redirect URL.
+		$success_redirect = add_query_arg( 'notification', 'some_notification_value', admin_url( 'success-redirect' ) );
+
+		$client->get_authentication_url( $success_redirect );
+		// No other way around this but to mock the Google_Site_Kit_Client
+		$google_client_mock = $this->getMockBuilder( 'Google\Site_Kit\Core\Authentication\Clients\Google_Site_Kit_Client' )
```

Comment:
This boilerplate copied from `test_authorize_user`... going for DAMP over DRY here in a test context but happy to refactor a bit if you think it's worthwhile.

Reply:
SGTM 👍